packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -671,24 +691,68 @@ async function createNewResourceGroup(
   return ok(response.name);
 }
 
-async function checkResourceGroupExistence(
+function handleRestError<T>(
+  restError: RestError,
+  resourceGroupName: string,
+  subscriptionId: string,
+  subscriptionName: string
+): Err<T, FxError> {
+  const subscriptionInfoString =
+    subscriptionId + (subscriptionName.length > 0 ? `(${subscriptionName})` : "");
+  if (restError.statusCode === 403) {
+    return err(
+      returnUserError(
+        new Error(
+          `Unauthorized to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}'. Please check your Azure subscription.`
+        ),
+        SolutionSource,
+        SolutionError.Unauthorized
+      )
+    );
+  } else {
+    // Avoid sensitive information like request headers in the error message.
+    const rawErrorString = JSON.stringify({
+      code: restError.code,
+      statusCode: restError.statusCode,
+      body: restError.body,
+      name: restError.name,
+      message: restError.message,
+    });
+    return err(
+      returnUserError(
+        new Error(
+          `Failed to check the existence of resource group '${resourceGroupName}' in subscription '${subscriptionInfoString}', raw error: '${rawErrorString}'`
+        ),
+        SolutionSource,
+        SolutionError.FailedToCheckResourceGroupExistence
+      )
```

Comment:
We have several `UserError` related to the resource group check, is that possible to wrap/abstract these kinds of `UserError` for better error handling/monitoring, I remember Huajie has talked about how to define the Error in Teams toolkit before.

Reply:
It makes sense. I will refactor it.